src/query/service/src/servers/http/middleware.rs

```diff
@@ -176,10 +179,29 @@ fn auth_by_header(
         }
     } else if value.as_bytes().starts_with(b"Bearer ") {
         match Bearer::decode(value) {
-            Some(bearer) => Ok(Credential::Jwt {
-                token: bearer.token().to_string(),
-                client_ip,
-            }),
+            Some(bearer) => {
+                let token = bearer.token().to_string();
+                if SessionClaim::is_databend_token(&token) {
+                    let (token_type, set_user) = if path == "/query" {
```

Comment:
imho it's not a common practice to distinguish tokens by url paths.

when we need refreshing tokens, we can simply pass the refresh token in the request body, and get the new tokens in the response.

Reply:
maybe another way is to encode the different types as prefix in the generated tokens: 

```
SessionClaim::is_databend_session_token(&token) 
SessionClaim::is_databend_refresh_token(&token) 
```

Reply:
@flaneur2020 you are right, I was planning to do this refactor later, let me do it now.

Reply:
done